lib/spack/spack/version.py

```diff
@@ -1284,18 +1284,15 @@ def fetcher(self):
 
     @property
     def repository_uri(self):
-        """
-        Identifier for git repos used within the repo and metadata caches.
+        (scheme, netloc, path, params, query, fragment) = urllib.parse.urlparse(self.pkg.git)
+        if scheme == "file":
+            if sys.platform == "win32":
+                path = path.lstrip("/")
 
-        """
-        try:
-            components = [
-                str(c).lstrip("/") for c in spack.util.url.parse_git_url(self.pkg.git) if c
-            ]
-            return os.path.join(*components)
-        except ValueError:
-            # If it's not a git url, it's a local path
-            return os.path.abspath(self.pkg.git)
+            return Path("file").joinpath(Path(path).relative_to(Path.home()))
```

Comment:
This is bc the `repository_uri` function as it was written before is poorly named. It used to return the pieces of a file uri concatenated together to form a path. An example would be `file/AppData/Local/Temp/pytest-of-markus.ferrell/pytest-634/test_install_commit0/git_repo` instead of the real file uri of `file:///C:/Users/markus.ferrell/AppData/Local/Temp/pytest-of-markus.ferrell/pytest-634/test_install_commit0/git_repo`. They would use that path segment to concatenate onto another path to find the real file uri.

The relative to home was to try and resolve long path issues by removing part of the repeated path. I agree that it's too fragile though. It seems to be causing failures for other platforms too.